PTGLdynamics/codes/plotting-scripts/compareContactPartnersOfResidues.py

```diff
@@ -0,0 +1,222 @@
+                        matches[key_2] = value_2     
+                
+        else:
+            break
+    return matches
+           
+
+########### configure logger ###########
+
+logging.basicConfig(format = "[%(levelname)s] %(message)s")
+            
+            
+########### not built-in imports ###########
+           
+########### command line parser ###########
+
+## create the parser
+cl_parser = argparse.ArgumentParser(description="Compares two csv files containing contacts between residues of different chains. Returns the number of changes in contact partners in a csv file.",
+                                    fromfile_prefix_chars="@")
+
+## add arguments
+
+# add mutually exclusive group for silent / verbose
+loudness = cl_parser.add_mutually_exclusive_group()
+loudness.add_argument('-s',
+                       '--silent',
+                       action='store_true',
+                       help='only print results to stdout, e.g., no status messages and meta information')
+
+loudness.add_argument('-v',
+                       '--verbose',
+                       action='store_true',
+                       help='print more to stdout, e.g., status messages and meta information')
+
+loudness.add_argument('-d',
+                      '--debug',
+                      action='store_true',
+                      help='print everything including debug information')
+
+# add command line arguments
+cl_parser.add_argument('--version',
+                       action='version',
+                       version='%(prog)s ' + version)
+
+cl_parser.add_argument('file_one',
+                       metavar = 'path',
+                       help='the first csv file containing the residue contacts for a residue.')
+
+cl_parser.add_argument('file_two',
+                       metavar = 'path',
+                       help='the second csv file containing the residue contacts for a residue.')
+
+cl_parser.add_argument('-p',
+                       '--outputdirectory',
+                       metavar = 'path',
+                       default = '',
+                       help = 'specify a path to your output files. Otherwise the current folder is used.')
+
+args = cl_parser.parse_args()
+
+
+########### check arguments ###########
+
+# assign log level
+log_level = logging.WARNING
+if (args.debug):
+    log_level = logging.DEBUG
+elif (args.verbose):
+    log_level = logging.INFO
+logging.getLogger().setLevel(log_level)
+
+# input file 1
+file1 = check_input_files(args.file_one)
+        
+# input file 2
+file2= check_input_files(args.file_two)
+        
+# output directory
+output_dir = check_dir_args(args.outputdirectory)
+
+########### vamos ###########
+
+log("Version " + version, "i")
+
+try:
+    residues_contacts_file1 = pickle.load(open("residues_contacts", "rb"))                                          
+except FileNotFoundError:
+    residues_contacts_file1 = read_contact_file_in(file1)
```

Comment:
I like the try-except block here and the error handling newly reading the file if it does not exist